tests/bose/test_unary_mapping.py

```diff
@@ -0,0 +1,637 @@
+        qubit_op.simplify(tol=1e-8)
+
+        expected_op = pauli_sentence(qml.Hamiltonian(result[0], result[1]))
+        expected_op.simplify(tol=1e-8)
+        assert qubit_op == expected_op
+
+
+@pytest.mark.parametrize(
+    "bose_op",
+    [
+        BoseWord({(0, 0): "-"}),
+        BoseSentence({BoseWord({(0, 0): "-"}): 1.0, BoseWord({(0, 1): "-"}): 1.0}),
+    ],
+)
+def test_return_unary_mapping_sum(bose_op):
+    """Test that the correct type is returned for unary mapping
+    when ps is set to False."""
+
+    qubit_op = unary_mapping(bose_op, ps=False)
+    assert isinstance(qubit_op, qml.ops.Sum)
+
+
+@pytest.mark.parametrize(
+    "bose_op",
+    [
+        BoseWord({(0, 0): "-"}),
+        BoseSentence({BoseWord({(0, 0): "-"}): 1.0, BoseWord({(0, 1): "-"}): 1.0}),
+    ],
+)
+def test_return_unary_mapping_ps(bose_op):
+    """Test that the correct type is returned for unary mapping
+    when ps is set to False."""
+
+    qubit_op = unary_mapping(bose_op, ps=True)
+    assert isinstance(qubit_op, qml.pauli.PauliSentence)
+
+
+@pytest.mark.parametrize(
+    ("bose_op, wire_map, result"),
+    [
+        (
+            BoseWord({(0, 0): "+"}),
+            {0: 1, 1: 2},
+            (
+                [0.25, -0.25j, 0.25j, (0.25 + 0j)],
+                [X(1) @ X(2), X(1) @ Y(2), Y(1) @ X(2), Y(1) @ Y(2)],
+            ),
+        ),
+        (
+            BoseSentence({BoseWord({(0, 0): "-"}): 1.0, BoseWord({(0, 1): "+"}): 1.0}),
+            {0: "a", 1: "b", 2: "c", 3: "d"},
+            (
+                [0.25, 0.25j, -0.25j, (0.25 + 0j), 0.25, -0.25j, 0.25j, (0.25 + 0j)],
+                [
+                    X("a") @ X("b"),
+                    X("a") @ Y("b"),
+                    Y("a") @ X("b"),
+                    Y("a") @ Y("b"),
+                    X("c") @ X("d"),
+                    X("c") @ Y("d"),
+                    Y("c") @ X("d"),
+                    Y("c") @ Y("d"),
+                ],
+            ),
+        ),
+    ],
+)
+def test_unary_mapping_wiremap(bose_op, wire_map, result):
+    """Test that the unary_mapping function returns the correct qubit operator."""
+    qubit_op = unary_mapping(bose_op, n_states=2, wire_map=wire_map, ps=True)
+    qubit_op.simplify(tol=1e-8)
+
+    expected_op = pauli_sentence(qml.Hamiltonian(result[0], result[1]))
+    expected_op.simplify(tol=1e-8)
+    assert qubit_op == expected_op
+
+
+def test_n_states_error_unary():
+    """Test that an error is raised if invalid number of states is provided."""
+    bw = BoseWord({(0, 0): "-"})
+    with pytest.raises(
+        ValueError, match="Number of allowed bosonic states cannot be less than 2, provided 0."
+    ):
+        unary_mapping(bw, n_states=0)
+
+
+def test_error_is_raised_for_incompatible_type():
+    """Test that an error is raised if the input is not a BoseWord or BoseSentence"""
+
+    with pytest.raises(ValueError, match="bose_operator must be a BoseWord or BoseSentence"):
+        unary_mapping(X(0))
```

Comment:
Also just realized this should be a TypeError instead. But it's not blocking just would be nice to fix before merge.